connection.go

```diff
@@ -67,28 +76,28 @@ func (c *fireboltConnection) queryContextInternal(ctx context.Context, query str
 		return nil, fmt.Errorf("multistatement is not allowed")
 	}
 
-	var rows rows.InMemoryRows
+	var rowsInst = c.makeRows(ctx)
+
 	for _, query := range queries {
 		if isSetStatement, err := processSetStatement(ctx, c, query); isSetStatement {
-			if err == nil {
-				rows.AppendResponse(types.QueryResponse{})
-				continue
-			} else {
-				return &rows, errorUtils.ConstructNestedError("statement recognized as an invalid set statement", err)
+			if err != nil {
+				return rowsInst, errorUtils.ConstructNestedError("statement recognized as an invalid set statement", err)
+			} else if err = rowsInst.AppendResponse(client.MakeResponse(nil, 200, nil, nil)); err != nil {
+				return rowsInst, errorUtils.ConstructNestedError("error during query execution", err)
```

Comment:
nit: not error during query execution, but error appending or processing response? I was trying to make sense on what is actually happening here for some time.

Reply:
Not sure I fully understand your comment. The error can happen in both query request and appending the response to rows. 
It's because inMemory read the response in the AppendResponse call, and can find a structured error there. While streaming rows just save the response stream and read it on demand
Maybe you can show the example of how do you want this changed?

Reply:
The thing is that parsing is only done for InMemory rows, StreamRows simply append the stream, so mentioning validation can make it confusing. I've added  comment for each function that would hopefully clarify it

Reply:
I feel like it's a design problem then. When reading through the code it's not obvious that append response method would throw execution errors. Unless we remember how this works (and in a few weeks who knows if we do) runtime errors might come as a surprise.